pkg/docker/containers-view.jsx

```diff
@@ -0,0 +1,447 @@
+                </ul>
+            </div>
+        );
+    }
+});
+
+var ContainerHeader = React.createClass({
+    getInitialState: function () {
+        return {
+            filter: 'running',
+            filterText: ''
+        }
+    },
+
+    filterChanged: function () {
+        if (this.props.onFilterChanged)
+            this.props.onFilterChanged(this.state.filter, this.state.filterText);
+    },
+
+    handleFilterChange: function (value) {
+        this.setState({ filter: value }, this.filterChanged);
+    },
+
+    handleFilterTextChange: function (event) {
+        this.setState({ filterText: this.refs.filterTextInput.value }, this.filterChanged);
+    },
+
+    render: function () {
+        return (
+            <div>
+                <Select.Select id="containers-containers-filter" initial="running" onChange={this.handleFilterChange}>
+                    <Select.SelectEntry data='all'>{_('All')}</Select.SelectEntry>
+                    <Select.SelectEntry data='running'>{_('Running')}</Select.SelectEntry>
+                </Select.Select>
+                <input type="text"
+                       id="containers-filter"
+                       ref="filterTextInput"
+                       className="form-control"
+                       placeholder={_('Type to filter…')}
+                       onChange={this.handleFilterTextChange} />
+            </div>
+        );
+    }
+});
+
+var ContainerDetails = React.createClass({
+    render: function () {
+        var container = this.props.container;
+        return (
+            <div className='listing-body'>
+                <dl>
+                    <dt>Id       </dt> <dd>{ container.Id }</dd>
+                    <dt>Created  </dt> <dd>{ container.Created }</dd>
+                    <dt>Image    </dt> <dd>{ container.Image }</dd>
+                    <dt>Commmand </dt> <dd>{ util.render_container_cmdline(container) }</dd>
+                    <dt>State    </dt> <dd>{ util.render_container_state(container.State) }</dd>
+                </dl>
+            </div>
+        );
+    }
+});
+
+var ContainerList = React.createClass({
+    getDefaultProps: function () {
+        return {
+            client: {},
+            onlyShowRunning: true,
+            filterText: ''
+        };
+    },
+
+    getInitialState: function () {
+        return {
+            containers: []
+        };
+    },
+
+    navigateToContainer: function (container) {
+        cockpit.location.go([ container.Id ]);
+    },
+
+    startContainer: function (container) {
+        this.props.client.start(container.Id).fail(util.show_unexpected_error);
+    },
+
+    stopContainer: function (container) {
+        this.props.client.stop(container.Id).fail(util.show_unexpected_error);
+    },
+
+    restartContainer: function (container) {
+        this.props.client.restart(container.Id).fail(util.show_unexpected_error);
+    },
+
+    deleteContainer: function (container) {
+        if (event.button !== 0)
```

Comment:
Weird, WebKit has a global `event`, which makes this work. Thanks for the catch. Fixed.